tests/test_ingest.py

```diff
@@ -25,38 +25,336 @@
 import tempfile
 import unittest
 
-import lsst.daf.butler as dafButler
+import lsst.log
 import lsst.daf.butler.tests as butlerTests
+from lsst.daf.butler import DatasetType, Butler, DataCoordinate, Config
+from lsst.daf.butler.registry import ConflictingDefinitionError
+from lsst.daf.butler.core.utils import getFullTypeName
 
+from lsst.obs.base.ingest_tests import IngestTestBase
+from lsst.obs.base.instrument_tests import DummyCam
 from lsst.obs.base import RawIngestTask
 
 
-TESTDIR = os.path.dirname(__file__)
+TESTDIR = os.path.abspath(os.path.dirname(__file__))
+INGESTDIR = os.path.join(TESTDIR, "data", "ingest")
 
 
-class RawIngestTestCase(unittest.TestCase):
+class DummyCamRawIngestTask(RawIngestTask):
+    """For DummyCam we ingest a different dataset type that can return
+    a non-Exposure."""
+
+    def getDatasetType(self):
+        """Return the DatasetType of the datasets ingested by this Task.
+        """
+        return DatasetType("raw_dict", ("instrument", "detector", "exposure"), "StructuredDataDict",
+                           universe=self.butler.registry.dimensions)
+
+
+class RawIngestTestCase(IngestTestBase, unittest.TestCase):
+    """Test ingest using JSON sidecar files."""
+
+    ingestDatasetTypeName = "raw_dict"
+    rawIngestTask = getFullTypeName(DummyCamRawIngestTask)
+    curatedCalibrationDatasetTypes = ()
+    ingestDir = TESTDIR
+    instrumentClassName = "lsst.obs.base.instrument_tests.DummyCam"
+    file = os.path.join(INGESTDIR, "sidecar_data", "dataset_1.yaml")
+    dataIds = [dict(instrument="DummyCam", exposure=100, detector=0)]
+
+    @property
+    def visits(self):
+        butler = Butler(self.root, collections=[self.outputRun])
+        return {
+            DataCoordinate.standardize(
+                instrument="DummyCam",
+                visit=100,
+                universe=butler.registry.dimensions
+            ): [
+                DataCoordinate.standardize(
+                    instrument="DummyCam",
+                    exposure=100,
+                    universe=butler.registry.dimensions
+                )
+            ]
+        }
+
+    def testWriteCuratedCalibrations(self):
+        """There are no curated calibrations in this test instrument"""
+        pass
+
+
+class RawIngestImpliedIndexTestCase(RawIngestTestCase):
+    """Test ingest using JSON index files."""
+    file = os.path.join(INGESTDIR, "indexed_data", "dataset_1.yaml")
+
+
+class RawIngestEdgeCaseTestCase(unittest.TestCase):
+    """Test ingest using non-standard approaches including failures."""
+
     @classmethod
     def setUpClass(cls):
-        """Create a new butler once only."""
+        butlerConfig = """
+datastore:
+  # Want to ingest real files so can't use in-memory datastore
+  cls: lsst.daf.butler.datastores.fileDatastore.FileDatastore
+"""
```

Comment:
check indenting here?

Reply:
Indenting looks fine to me. What is the problem? It's a yaml string so it normally wouldn't be indented.